core/src/main/scala/com/metamx/tranquility/beam/ClusteredBeam.scala

```diff
@@ -178,67 +213,78 @@ class ClusteredBeam[EventType: Timestamper, InnerBeamType <: Beam[EventType]](
     t => theImplicit(t).withZone(DateTimeZone.UTC)
   }
 
-  private[this] def beam(timestamp: DateTime, now: DateTime): Future[Beam[EventType]] = {
-    val bucket = tuning.segmentBucket(timestamp)
+  private[this] def beam(beamPair :(Interval, Option[Beam[EventType]]), now: DateTime): Future[Beam[EventType]] = {
     val creationInterval = new Interval(
-      tuning.segmentBucket(now - tuning.windowPeriod).start.millis,
-      tuning.segmentBucket(Seq(now + tuning.warmingPeriod, now + tuning.windowPeriod).maxBy(_.millis)).end.millis,
-      ISOChronology.getInstanceUTC
+      tuning.segmentBucket(now - tuning.windowPeriod).start,
+      tuning.segmentBucket(Seq(now + tuning.warmingPeriod, now + tuning.windowPeriod).maxBy(_.millis)).end
     )
-    val windowInterval = new Interval(
-      tuning.segmentBucket(now - tuning.windowPeriod).start.millis,
-      tuning.segmentBucket(now + tuning.windowPeriod).end.millis,
-      ISOChronology.getInstanceUTC
-    )
-    val futureBeamOption = beams.get(timestamp.millis) match {
+
+    val futureBeamOption = beamPair match {
       case _ if !open => Future.value(None)
-      case Some(x) if windowInterval.overlaps(bucket) => Future.value(Some(x))
-      case Some(x) => Future.value(None)
-      case None if timestamp <= localLatestCloseTime => Future.value(None)
-      case None if !creationInterval.overlaps(bucket) => Future.value(None)
-      case None =>
-        // We may want to create new merged beam(s). Acquire the zk mutex and examine the situation.
-        // This could be more efficient, but it's happening infrequently so it's probably not a big deal.
+      case (interval, Some(foundBeam)) if foundBeam.getInterval().get.end + tuning.windowPeriod <= now => Future.value(None)
+      case (interval, Some(foundBeam)) => Future.value(Some(foundBeam))
+      //case (interval, None) if interval.start <= localLatestCloseTime => Future.value(None)
```

Comment:
So earlier it was needed because we were not creating beams in past but now I am not sure whether to follow the same strategy or not. For example, if we have tasks with 5 minute seg gran and 1 minute window period. Current time is 1:43,  currently running task is for interval 1:40-1:45 and there was only one past task for interval 1:35-1:40. 

Now if user changes the segment gran to 15 mins and then tranquility receives the data with timestamp say 1:34 then what should be done ? Ideally speaking, 1:30 - 1:45 falls in a 15 min segment and there is not segment for interval 1:30-1:35 that can handle event with ts as 1:34. This code will create a task that can handle this event and will have interval 1:30-1:35. However, without this change, the event would have been rejected as the `localLatestCloseTime` would be 1:35 and new task interval would be 1:30. 

I do not know if this should be done or not ? The comment was left intentionally so that I remember to confirm.